blueflood-http/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/HttpEventsQueryHandler.java

```diff
@@ -0,0 +1,74 @@
+package com.rackspacecloud.blueflood.outputs.handlers;
+
+import com.rackspacecloud.blueflood.http.HTTPRequestWithDecodedQueryParams;
+import com.rackspacecloud.blueflood.http.HttpRequestHandler;
+import com.rackspacecloud.blueflood.http.HttpResponder;
+import com.rackspacecloud.blueflood.io.GenericElasticSearchIO;
+import com.rackspacecloud.blueflood.io.Constants;
+
+import com.rackspacecloud.blueflood.utils.DateTimeParser;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.jboss.netty.buffer.ChannelBuffers;
+import org.jboss.netty.channel.ChannelHandlerContext;
+import org.jboss.netty.handler.codec.http.*;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+
+public class HttpEventsQueryHandler implements HttpRequestHandler {
+    private static final Logger log = LoggerFactory.getLogger(HttpEventsQueryHandler.class);
+    private GenericElasticSearchIO searchIO;
+
+    public HttpEventsQueryHandler(GenericElasticSearchIO searchIO) {
+        this.searchIO = searchIO;
+    }
+
+
+    @Override
+    public void handle(ChannelHandlerContext ctx, HttpRequest request) {
+        final String tenantId = request.getHeader("tenantId");
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        String responseBody;
+        try {
+            HTTPRequestWithDecodedQueryParams requestWithParams = (HTTPRequestWithDecodedQueryParams) request;
+            Map<String, List<String>> params = requestWithParams.getQueryParams();
+
+            parseDateFieldInQuery(params, "from");
+            parseDateFieldInQuery(params, "until");
+
+            List<Map<String, Object>> searchResult = searchIO.search(tenantId, params);
+            responseBody = objectMapper.writeValueAsString(searchResult);
+        }
+        catch (Exception e) {
+            log.error(String.format("Exception %s", e.toString()));
+            responseBody = String.format("Error: %s", e.getMessage());
+        }
+
+        sendResponse(ctx, request, responseBody, HttpResponseStatus.OK);
```

Comment:
TODO: Even if there is an exception, this line returns an OK ResponseCode. For example if searchIO is null, the NPE gets masked and does not return a 500. Fix!